### PR TITLE
UI: [VAULT-21521] Initial config-ui engine and routes set up

### PR DESCRIPTION
**Pull request by kiannaquach**
The engine and route name changed a bit from the RFC. Decided to name the engine `config-ui` to have a posterity with the API endpoint name.

**Engine name:** config-ui
**Routes in config-ui:**
```
  this.route('messages', function () {
    this.route('create');
    this.route('message', { path: '/:id' }, function () {
      this.route('details');
      this.route('edit');
    });
  });
```

### CODE REVIEW

File: ui/lib/config-ui/addon/engine.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
```
Comment (zofskeez) [inline ui/lib/config-ui/addon/engine.js:3]:
```suggestion
 * SPDX-License-Identifier: BUSL-1.1
```

File: ui/lib/config-ui/addon/resolver.js
Comment (zofskeez) [inline ui/lib/config-ui/addon/resolver.js:1]: This file actually isn't necessary if you aren't extending the resolver.

File: ui/lib/config-ui/addon/routes.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
```
Comment (zofskeez) [inline ui/lib/config-ui/addon/routes.js:3]:
```suggestion
 * SPDX-License-Identifier: BUSL-1.1
```

File: ui/lib/config-ui/config/environment.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
```
Comment (zofskeez) [inline ui/lib/config-ui/config/environment.js:3]:
```suggestion
 * SPDX-License-Identifier: BUSL-1.1
```

**Review by zofskeez — CHANGES_REQUESTED** [review]
Just a couple minor changes but everything looks good!

File: ui/lib/config-ui/addon/routes.js (hunk 2 above)
Comment (kiannaquach) [reply ui/lib/config-ui/addon/routes.js:3]: Ahh good catch!

**Review by zofskeez — APPROVED** [review]
🎉

**Pull request merged**